io/snmp/snmp.js

```diff
@@ -51,6 +51,53 @@ module.exports = function(RED) {
     }
     RED.nodes.registerType("snmp",SnmpNode);
 
+ function SnmpSNode(n) {
+        RED.nodes.createNode(this, n);
+        this.community = n.community;
+        this.host = n.host;
+        this.version = (n.version === "2c") ? snmp.Version2c : snmp.Version1;
+        var node = this;
+
+        this.on("input", function (msg) {
+            var host = node.host || msg.host;
+            var community = node.community || msg.community;
+            var oids = msg.oids;
+            node.log(oids + ' ' + host + ' ' + msg.host+ community + msg.community);
+            if (oids) {
+                node.session = snmp.createSession(host, community, { version: node.version });
+                node.session.set(oids, function (error, varbinds) {
+                    if (error) {
+                        node.error(error.toString(), msg);
+                    }
+                    else {
+                        for (var i = 0; i < varbinds.length; i++) {
+                            if (snmp.isVarbindError(varbinds[i])) {
+                                node.error(snmp.varbindError(varbinds[i]), msg);
+                            }
+                            else {
+                                if (varbinds[i].type == 4) { varbinds[i].value = varbinds[i].value.toString(); }
+                                varbinds[i].tstr = snmp.ObjectType[varbinds[i].type];
+                                //node.log(varbinds[i].oid + "|" + varbinds[i].tstr + "|" + varbinds[i].value);
+                            }
+                        }
+                        msg.oid = oids;
+                        msg.payload = varbinds;
+                        node.send(msg);
```

Comment:
Ok, I agree that was for error feedback but we won't manage errors within node output, so let's agree on your code.
At the moment varbinds must contain oids in JSON syntax   [ { oid: "1.3.6.1.2.1.1.5.0", type: 4, value: "host1"}  with type value in int according to net-snmp lib : 
```
var ObjectType = {
	1: "Boolean",
	2: "Integer",
	4: "OctetString",
	5: "Null",
	6: "OID",
	64: "IpAddress",
	65: "Counter",
	66: "Gauge",
	67: "TimeTicks",
	68: "Opaque",
	70: "Counter64",
	128: "NoSuchObject",
	129: "NoSuchInstance",
	130: "EndOfMibView"
}; 
```
A helper is indeed a good idea to make it more explicit.